duckduckgo/iOS
# Add an “Open VPN Settings” quick action

Task/Issue URL: https://app.asana.com/0/414235014887631/1205784963805147/f
Tech Design URL:
CC: @graeme @diegoreymendez 

**Description**:

This PR adds an iOS VPN quick action.



**Steps to test this PR**:
1. Launch a clean debug build of the app without any NetP access (i.e., no auth token) and verify that **no quick action appears** when you long-press the DDG app icon
2. Mark yourself as an internal user and authenticate with NetP
3. Dismiss the settings screen
4. Go back to the iOS Home Screen and long-press the DDG icon, verify that the "Open VPN Settings" option appears and correctly takes you to the VPN settings screen

Next, test that opening the quick action doesn't re-present UI needlessly.

1. Open the settings screen (but NOT the NetP screen)
2. Dismiss the app and use the Open VPN action
3. Check that the NetP screen is instantly presented
4. When you have the NetP screen open, dismiss the app and use the Open VPN action
5. Check that nothing happens, as the NetP screen is already open



**Copy Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
